### PR TITLE
OCPBUGS-42298: Remove ClusterTask dependency in console from Pipelines 1.17

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Fixes**: 
https://issues.redhat.com/browse/OCPBUGS-42298

**Solution Description**: 
ClusterTask has been deprecated and will be removed in Pipelines Operator 1.17

So to support this,

1. Removed ClusterTask list page in search menu if Pipelines Operator >= 1.17
2. Removed ClusterTask list page tab in Tasks navigation menu if Pipelines Operator >= 1.17
3. ClusterTask is removed from quick search in Pipelines builder if Pipelines Operator >= 1.17
  

**Unit test coverage report**: 
Screenshot 2024-10-01 at 4 33 49 PM

**Test setup:**

    1. Install Pipelines operator
    2. If Pipeline operator is 1.16 or lower, then it should work as existing behaviour, used should see ClusterTask in Task navigation menu and Pipeline builder page
    3. Since Pipelines 1.17 is not installed, for testing purpose, make the below changes to test in `useIsPipelineOperatorVersion_1_17` function in `pipeline-operator.ts` file,
```
    const IS_PIPELINE_OPERATOR_VERSION_1_17 =
    operatorVersion?.major === 1 && operatorVersion?.minor >= 17;
```
to
    
```
    const IS_PIPELINE_OPERATOR_VERSION_1_17 =
    operatorVersion?.major === 1 && operatorVersion?.minor >= 15;
```
    

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/pipelines/utils/pipeline-operator.ts [hunk 1]
```diff
@@ -101,3 +102,12 @@ export const useIsTektonV1VersionPresent = (setFeatureFlag: SetFeatureFlag) => {
   const isTektonV1VersionPresent = operatorVersion?.major === 1 && operatorVersion?.minor >= 11;
   setFeatureFlag(FLAG_TEKTON_V1_ENABLED, isTektonV1VersionPresent);
 };
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const useIsPipelineOperatorVersion_1_17 = (setFeatureFlag: SetFeatureFlag) => {
```
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply frontend/packages/pipelines-plugin/src/components/pipelines/utils/pipeline-operator.ts]: Updated the PR.

**Review by logonoff — COMMENTED** [review]
/lgtm

tested on a 4.17.1 cluster, working fine locally: clustertask shows up when operator version is set to 1.16.0, and is removed when i manually change the operator version to 1.17.0

**Comment by lokanandaprabhu** [comment]
Rebased the PR because of merge conflict.

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Comment by lokanandaprabhu** [comment]
Rebased the PR because of merge conflict.

**Pull request merged**